src/main/java/ch/njol/skript/effects/EffEquip.java

```diff
@@ -94,82 +104,80 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Override
 	@SuppressWarnings("deprecation")
 	protected void execute(Event event) {
-		ItemType[] ts = types.getArray(event);
+		ItemType[] ts = itemTypes.getArray(event);
 		for (LivingEntity en : entities.getArray(event)) {
 			if (SUPPORTS_STEERABLE && en instanceof Steerable) {
 				for (ItemType it : ts) {
 					if (SADDLE.isOfType(it.getMaterial())) {
-						((Steerable) en).setSaddle(true);
+						((Steerable) en).setSaddle(equip);
 					}
 				}
+				continue;
 			} else if (en instanceof Pig) {
 				for (ItemType t : ts) {
 					if (t.isOfType(Material.SADDLE)) {
-						((Pig) en).setSaddle(true);
+						((Pig) en).setSaddle(equip);
 						break;
 					}
 				}
 				continue;
 			} else if (SUPPORTS_LLAMAS && en instanceof Llama) {
-				LlamaInventory invi = ((Llama) en).getInventory();
+				LlamaInventory inv = ((Llama) en).getInventory();
 				for (ItemType t : ts) {
 					for (ItemStack item : t.getAll()) {
 						if (CARPET.isOfType(item)) {
-							invi.setDecor(item);
+							inv.setDecor(equip ? item : new ItemStack(Material.AIR));
```

Comment:
Instead of calling `new ItemStack(Material.AIR)` for each condition, just create a field for it below the static block